core/dbt/common/constraints.py

```diff
@@ -0,0 +1,43 @@
+from dataclasses import dataclass, field
```

Comment:
Should we consider constraints a 'contract'? If so we should move it to common/contracts

Reply:
@MichelleArk What do you think about this?

Reply:
agree - common/contracts makes sense as a home for these based on the existing patterns we have in dbt & dbt/adapters